packages/react-devtools-shared/src/utils.js

```diff
@@ -510,7 +510,11 @@ export function getDataType(data: Object): DataType {
     return 'react_element';
   }
 
-  if (typeof HTMLElement !== 'undefined' && data instanceof HTMLElement) {
+  if (
+    (typeof HTMLElement !== 'undefined' && data instanceof HTMLElement) ||
+    // cross-realm HTMLElement?
+    /^\[object HTML.+Element\]$/.test(Object.prototype.toString.call(data))
```

Comment:
As far as I can tell this is only needed for the shell at the moment. Without this change

```jsx
function Example() {
  const ref = React.useRef();

  return <React.Fragment><div ref={ref} /><SomeComponent foo={ref} /></React.Fragment>
}
```

would also crash if one would attempt to inspect `SomeComponent` since `foo.current` would not be considered an `HTMLElement`. I can add the described case to the existing shell to make sure we don't regress when removing the special casing of `ref`.